### PR TITLE
[ci skip] Updating edge rails guides to include `on_weekday?` on Date, Time and DateTime

### PR DESCRIPTION
**Pull request by exagil**

### CODE REVIEW

**Comment by prathamesh-sonpatki** [comment]
r? @vipulnsward 

@chi6rag Please add `[ci skip]` going forward to changes only to documentation so that Travis won't run for such doc changes.

**Comment by chi6rag** [comment]
Added

**Comment by prathamesh-sonpatki** [comment]
@chi6rag We need to add it to commit message not PR title. It's fine to keep it as it is now that CI has kicked off, please add it next time :smile:

**Comment by chi6rag** [comment]
ah! got it! thanks!

File: guides/source/active_support_core_extensions.md [hunk 1]
```diff
@@ -3078,7 +3078,7 @@ INFO: The following calculation methods have edge cases in October 1582, since d
 
 #### `Date.current`
 
-Active Support defines `Date.current` to be today in the current time zone. That's like `Date.today`, except that it honors the user time zone, if defined. It also defines `Date.yesterday` and `Date.tomorrow`, and the instance predicates `past?`, `today?`, and `future?`, all of them relative to `Date.current`.
+Active Support defines `Date.current` to be today in the current time zone. That's like `Date.today`, except that it honors the user time zone, if defined. It also defines `Date.yesterday` and `Date.tomorrow`, and the instance predicates `past?`, `today?`, `future?`, and `on_weekday?`, all of them relative to `Date.current`.
```
Comment (vipulnsward) [inline guides/source/active_support_core_extensions.md]: `on_weekday?` and `on_weekend?`

File: guides/source/active_support_core_extensions.md [hunk 2]
```diff
@@ -3467,6 +3467,7 @@ years_ago
 years_since
 prev_year (last_year)
 next_year
+on_weekday?
```
Comment (vipulnsward) [inline guides/source/active_support_core_extensions.md]: I was going through the complete list in this file. Can you update this to have some other missing methods if any?

File: guides/source/active_support_core_extensions.md [hunk 3]
```diff
@@ -3653,6 +3654,7 @@ years_ago
...
```
Comment (vipulnsward) [inline guides/source/active_support_core_extensions.md]: ditto

**Comment by chi6rag** [comment]
updated!

File: guides/source/active_support_core_extensions.md [hunk 4]
```diff
@@ -3078,7 +3078,7 @@ INFO: The following calculation methods have edge cases in October 1582, since d
 
 #### `Date.current`
 
-Active Support defines `Date.current` to be today in the current time zone. That's like `Date.today`, except that it honors the user time zone, if defined. It also defines `Date.yesterday` and `Date.tomorrow`, and the instance predicates `past?`, `today?`, and `future?`, all of them relative to `Date.current`.
+Active Support defines `Date.current` to be today in the current time zone. That's like `Date.today`, except that it honors the user time zone, if defined. It also defines `Date.yesterday` and `Date.tomorrow`, and the instance predicates `past?`, `today?`, `future?`, `on_weekday?` and `on_weekend?`, all of them relative to `Date.current`.
```
Comment (vipulnsward) [inline guides/source/active_support_core_extensions.md]: We have no changes in TimeWithZone. Let me verify its effects.

**Comment by chi6rag** [comment]
I think we could also add `on_weekday?` and `on_weekend?` methods on `TimeWithZone`, whatsay?

File: guides/source/active_support_core_extensions.md (hunk 4 above)
Comment (vipulnsward) [inline guides/source/active_support_core_extensions.md]: These get added as well.

**Comment by vipulnsward** [comment]
@chi6rag can you squash your commits, then this is ready to go.

**Comment by chi6rag** [comment]
good to go!

**Comment by vipulnsward** [comment]
Thanks @chi6rag !

**Pull request merged**